spikeextractors/extraction_tools.py

```diff
@@ -325,33 +350,57 @@ def write_to_binary_dat_format(recording, save_path=None, file_handle=None,
             traces.tofile(file_handle)
     else:
         # chunk size is not None
-        n_sample = recording.get_num_frames()
-        n_chunk = n_sample // chunk_size
-        if n_sample % chunk_size > 0:
-            n_chunk += 1
-        if verbose:
-            chunks = tqdm(range(n_chunk), ascii=True, desc="Writing to binary .dat file")
+        num_frames = recording.get_num_frames()
+        num_channels = recording.get_num_channels()
+
+        # chunk_size = num_bytes_per_chunk / num_bytes_per_frame
+        chunks = divide_recording_into_time_chunks(
+            num_frames=num_frames,
+            chunk_size=chunk_size,
+            padding_size=0
+        )
+        n_chunk = len(chunks)
+
+        if verbose and n_jobs == 1:
+            chunks_loop = tqdm(range(n_chunk), ascii=True, desc="Writing to binary .dat file")
         else:
-            chunks = range(n_chunk)
+            chunks_loop = range(n_chunk)
         if save_path is not None:
-            with save_path.open('wb') as f:
-                for i in chunks:
-                    traces = recording.get_traces(start_frame=i * chunk_size,
-                                                  end_frame=min((i + 1) * chunk_size, n_sample))
-                    if dtype is not None:
-                        traces = traces.astype(dtype)
-                    if time_axis == 0:
-                        traces = traces.T
-                    f.write(traces.tobytes())
+            if n_jobs == 1:
+                with save_path.open('wb') as f:
+                    for i in chunks_loop:
```

Comment:
Please use _write_dat_one_chunk here to avoid mismatch in implementation if future change in _write_dat_one_chunk